packages/Frontend/src/containers/AuthenticateContainer.jsx

```diff
@@ -23,47 +22,86 @@ class AuthenticateContainer extends Component {
     super(props);
     this.state = {
       email: '',
-      error: '',
+      error: false,
       loading: false,
       password: '',
     };
   }
 
+  componentDidMount = () => {
+    this.setState({ loading: true }, () => {
+      auth().getRedirectResult().then((result) => {
+        this.setState({ loading: false });
+        console.log('success', result);
+      }, (error) => {
+        this.setState({ loading: false });
+        // The provider's account email, can be used in case of
+        // auth/account-exists-with-different-credential to fetch the providers
+        // linked to the email:
+        console.error(error);
+        console.error(error.message);
+        console.error(error.code);
+        // In case of auth/account-exists-with-different-credential error,
+        // you can fetch the providers using this:
+        if (error.code === 'auth/account-exists-with-different-credential') {
+          auth().fetchProvidersForEmail(error.email).then((providers) => {
+            console.log('other providers', providers);
+          // The returned 'providers' is a list of the available providers
+          // linked to the email address. Please refer to the guide for a more
+          // complete explanation on how to recover from this error.
+          });
+        }
+      });
+    });
+  };
+
   updateFormByKey = key => ({ target }) => {
     this.setState({ [key]: target.value });
   };
 
-  authenticate = () => {
+  authenticate = (provider) => {
+    if (provider === 'google') {
+      return this.authenticateWithProvider(new auth.GoogleAuthProvider());
+    }
+    if (provider === 'facebook') {
+      return this.authenticateWithProvider(new auth.FacebookAuthProvider());
+    }
     if (this.props.match.url === signInRoute) {
       return this.signIn();
     }
     return this.signUp();
   }
 
   signIn = () => {
-    const { email, password } = this.state;
-    this.setState({ error: '', loading: true }, () => {
-      Auth.signIn(email, password)
-        .then(() => Auth.currentAuthenticatedUser())
-        .then(user => this.props.login(user))
-        .catch(error => this.setState({ error }))
-        .finally(() => this.setState({ loading: false }));
+    this.setState({ loading: true, error: false }, () => {
```

Comment:
This is the way you handle logging in with an email and password via firebase. I got it from part of this guide: https://firebase.google.com/docs/auth/web/password-auth#sign_in_a_user_with_an_email_address_and_password